### ISSUE TITLE
Multipass ignores configured hypervisor settings and defaults to hyper-v on windows

### ISSUE THREAD

**Issue by demdante** [canonical/multipass#2179]
**Describe the bug**
I just recently installed the latest version of multipass on windows and, despite the fact that I configured virtualbox as the hypervisor to use during the install, when I run multipass start, I get this error message:

```
 multipass start
start failed: The Hyper-V Windows feature is disabled. Please enable by using the following
command in an Administrator Powershell and reboot:
Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All
```  

I can only assume that the reason I am seeing this error message is because multipass is ignoring my hypervisor configuration choice and defaulting to hyper-v, as enabling hyper-v on a windows machine will result in virtualbox no longer functioning at all:

![image](https://user-images.githubusercontent.com/64720591/126222533-39ce1c15-fb5a-402c-93f7-9339c310d1b5.png)

So, basically the virtualbox hypervisor is no longer usable, as multipass is requiring that we enable hyper-v first, which prevents virtualbox from working.

**To Reproduce**
How, and what happened?
1. On a windows 10 machine with virtualbox installed, virtualbox configured as the hypervisor, and hyper-v not enabled, run: `multipass start`

**Expected behavior**
Multipass should provision a linux vm using the virtualbox hypervisor

**Logs**

See attached

[logs.zip](https://github.com/canonical/multipass/files/6843786/logs.zip)

**Additional info**
 - OS: Windows 10 x64 Enterprise
- `multipass version`  1.7.0
- `multipass info --all`

```
C:\WINDOWS\system32>multipass info --all


C:\WINDOWS\system32>
```

That command returned nothing.

**Additional context**

**Comment by Saviq** [comment]
@demdante you can change the selected hypervisor with `multipass set local.driver=virtualbox` in an administrative shell. We'll investigate the installer to see what could have gone wrong there.

**Comment by Saviq** [comment]
Also, VirtualBox and Hyper-V do coexist OK, but preexisting VMs may not work indeed.

**Comment by demdante** [comment]
@Saviq Well, yeah, they can technically exist on the same machine.  Like I said, it just breaks virtualbox.  You can't use existing vms or create new ones, which makes virtualbox little more than a waste a hard drive space.  Also, I tried running the command you mentioned in your first post and, despite the fact that the command prompt clearly indicates it is running with administrative permissions, I get this:

![image](https://user-images.githubusercontent.com/64720591/126225639-0a952032-96c3-4d92-a61c-a1d026740d29.png)

I am guessing this is probably why the `multipass info --all` command didn't return any output.

**Comment by Saviq** [comment]
> @Saviq Well, yeah, they can technically exist on the same machine. Like I said, it just breaks virtualbox. You can't use existing vms or create new ones, which makes virtualbox little more than a waste a hard drive space.

That's not true, you _can_ run VirtualBox VMs with Hyper-V enabled. VirtualBox will actually _use_ Hyper-V then, but it will be transparent to you. VMs created prior may not work. Anyway, that's off topic :)

> Also, I tried running the command you mentioned in your first post and, despite the fact that the command prompt clearly indicates it is running with administrative permissions, I get this:
> I am guessing this is probably why the `multipass info --all` command didn't return any output.

Oh that's interesting. `info --all` returns nothing because there's no instances launched, that's fine. But why would an administrative console not have the necessary privileges...

In just ran a fresh installation and I can see the config set properly in the installer:

```plain
PS C:\Windows\System32\config\systemprofile\AppData\Local\multipassd> cat .\multipassd.conf
[General]
local.driver=virtualbox
```

Can you check that location for whether you can access it?

**Comment by demdante** [comment]
@Saviq I was just about to message you with that information you just found.  Here is the thing: why is multipass storing anything under system32?  That folder is extremely restrictive in regards to the kinds of applications which have access to it.  This is not going to be an issue for just me specifically, but for all windows users.  Sure, I could go into the folder and modify the permissions to allow multipass to read/write in the folder, but that is not really an acceptable solution here.  The solution is to not write in the system32 folder in the first place.

**Comment by Saviq** [comment]
@demdante we do want to move it to ProgramData indeed (#1112), but it's working just fine on most installs.

You can use the `MULTIPASS_STORAGE` environment variable (#1789) to change that location, btw.

**Comment by Saviq** [comment]
@demdante if you had some custom permissions on there, that might be why the installer couldn't write the config there, either.

Did you manage to resolve your issue?

**Comment by ricab** [comment]
I also observe the daemon not honoring the driver setting. Explicitly setting it appears to fix it. This on an admin shell:

```
PS C:\Windows\system32> multipass get local.driver
virtualbox
PS C:\Windows\system32> multipass networks
networks failed: The Hyper-V Windows feature is disabled. Please enable by using the following
command in an Administrator Powershell and reboot:
Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All
PS C:\Windows\system32> multipass set local.driver=virtualbox
PS C:\Windows\system32> multipass networks
Name        Type      Description
Ethernet 2  ethernet  Red Hat VirtIO Ethernet Adapter
```

**Comment by ricab** [comment]
This is a race in the installer. Restarting the daemon should be enough to workaround until it's fixed.
